src/Opw.PineBlog.EntityFrameworkCore/BlogEntityDbContext.cs

```diff
@@ -23,5 +31,162 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 entityType.SetTableName($"PineBlog_{entityType.GetTableName()}");
             }
         }
+
+        public Task<Result<int>> AddPostAsync(Post post, CancellationToken cancellationToken)
+        {
+            Posts.Add(post);
+            return SaveChangesAsync(cancellationToken);
+        }
+
+        public Task<Author> GetAuthorByUsernameAsync(string username)
+        {
+            return Authors.SingleOrDefaultAsync(a => a.UserName.Equals(username));
+        }
+
+        public Task<Result<int>> DeletePostAsync(Post post, CancellationToken cancellationToken)
+        {
+            Posts.Remove(post);
+            return SaveChangesAsync(true, cancellationToken);
+        }
+
+        public async Task<IList<Post>> GetPostListAsync(bool includeUnpublished, Pager pager, string category, PineBlogOptions options, CancellationToken cancellationToken)
```

Comment:
A repository should not have any business logic in it. So the creating of the `Pager` should be done in the `Command` or `Query`.

Reply:
Yes agreed. I do see that all `Pager`s are created in `Handler`s. However, each individual repository's underlying data access would need to handle pagination differently.

Reply:
The `skip` and `take` counts need to be params for the repository methods then